typhon/plugins/core.py

```diff
@@ -0,0 +1,198 @@
+            # was not directly imported was to avoid an unneeded dependency
+            # Parse the classname, arguments and keywords from the address
+            # First assume the form {class}|{args}|{kwargs}
+            try:
+                cls, args, kwargs = address.split('|', 2)
+            # Keyword arguments are optional i.e {class}|{args}
+            except ValueError:
+                cls, args = address.split('|', 1)
+                kwargs = None
+            # Import the class
+            mod, cls = cls.rsplit('.', 1)
+            if mod in sys.modules:
+                mod = sys.modules[mod]
+            else:
+                logger.debug("Importing %s", mod)
+                mod = importlib.import_module(mod)
+            # Grab our device
+            cls = getattr(mod, cls)
+            # Format arguments
+            if args:
+                args = args.split(',')
+            else:
+                args = list()
+            # Format keywords
+            if kwargs:
+                kwargs = dict([pair.split('=', 1)
+                               for pair in kwargs.split(',')])
+            else:
+                kwargs = dict()
+            # Create our object
+            logger.debug("Instantiating %s ...", cls.__name__)
+            self.obj = cls(*args, **kwargs)
+
+    @classmethod
+    def from_object(cls, obj):
+        """
+        Create a Connection object with an already instantiated object
+
+        Parameters
+        ----------
+        obj : object
+        """
+        return cls(None, None, preassembled=obj)
+
+
+class SignalConnection(ClassConnection):
+    """
+    Connection to monitor an Ophyd Signal
+
+    This is meant as a generalized connection to any type of Ophyd Signal. It
+    handles reporting new values to listeners as well as pushing new
+    values that users update in the PyDM interface back to the underlying
+    signal
+
+    Attributes
+    ----------
+    obj : ophyd.Signal
+        Stored signal object
+
+    Example
+    -------
+    .. code:: python
+
+        conn = ClassConnection(sig://ophyd.Signal|name=Test,)
+                               ophyd.Signal|name=Test)
+    """
+    supported_types = [int, float, str, np.ndarray]
+
+    def __init__(self, *args, **kwargs):
+        # Create Signal
+        super().__init__(*args, **kwargs)
+        # Subscribe to updates from Ophyd
+        self.obj.subscribe(self.send_new_value, event_type=self.obj.SUB_VALUE)
+        self.signal_type = None
+
+    @pyqtSlot(int)
+    @pyqtSlot(float)
+    @pyqtSlot(str)
+    @pyqtSlot(np.ndarray)
+    def put_value(self, new_val):
+        """
+        Pass a value from the UI to Signal
+
+        We are not guaranteed that this signal is writeable so catch exceptions
+        if they are created
+        """
+        try:
+            self.obj.put(new_val)
+        except Exception as exc:
+            logger.exception("Unable to put %s to %s", new_val, self.obj.name)
```

Comment:
Consider `%r` for `new_val`

@hhslepicka is logging in PyDM moving along yet? Would be great to have this info propagated back to the user outside of just the terminal.

Reply:
@klauer PyDM at master sets up the logger, we still need to make the usage of logging the default at all code instead of prints... There is still work to do but it works.

Reply:
Thanks @klauer! I know that this will use `repr` instead of `str` but what do you expect from `str` that you think will not be ideal? Just curious as I usually both interchangeably.

Reply:
Sorry for the late reply - I like `repr` here because then the exception message makes sense regardless of the value `new_val` contained. Strings, for example, get wrapped in quotes with `repr` and not with `str`. For user types, it assumes your classes have set `__repr__` and `__str__` appropriately, of course...